packages/server-wallet/src/wallet/store.ts

```diff
@@ -258,28 +262,43 @@ export class Store {
   }
 
   async addSignedState(
+    channelId: string,
     maybeChannel: Channel | undefined,
-    signedState: SignedStateWithHash,
+    wireSignedState: WireSignedState,
     tx: Transaction // Insist on a transaction because validateTransitionWIthEVM requires it
   ): Promise<Channel> {
-    const channelId = calculateChannelId(signedState);
     const timer = timerFactory(this.timingMetrics, `add signed state ${channelId}`);
 
-    await timer('validating signatures', async () => recoverParticipantSigners(signedState));
+    await timer('validating signatures', async () =>
+      recoverParticipantSigners(wireSignedState, channelId)
+    );
 
     const channel =
-      maybeChannel || (await timer('get channel', async () => getChannel(signedState, tx)));
+      maybeChannel || (await timer('get channel', async () => getChannel(channelId, tx)));
 
     if (!this.skipEvmValidation && channel.supported) {
       const {supported} = channel;
+      const supportedNitroState = {
```

Comment:
Why not use `wireStateToNitroState` here?

Reply:
Because `supported` is not a `WireState`. But your point still stands for `toNitroState`, which I should have used. 17f06b8